src/function/cast/struct_cast.cpp

```diff
@@ -9,10 +9,17 @@ unique_ptr<BoundCastData> StructBoundCastData::BindStructToStructCast(BindCastIn
 	vector<BoundCastInfo> child_cast_info;
 	auto &source_child_types = StructType::GetChildTypes(source);
 	auto &result_child_types = StructType::GetChildTypes(target);
+
 	if (source_child_types.size() != result_child_types.size()) {
 		throw TypeMismatchException(source, target, "Cannot cast STRUCTs of different size");
 	}
 	for (idx_t i = 0; i < source_child_types.size(); i++) {
+		auto target_is_unnamed = StructType::IsUnnamed(target, i);
+		auto source_is_unnamed = StructType::IsUnnamed(source, i);
+		if (!target_is_unnamed && !source_is_unnamed &&
+		    StringUtil::Lower(source_child_types[i].first) != StringUtil::Lower(result_child_types[i].first)) {
```

Comment:
Slight nit, you can use `StringUtil::CIEquals` instead of casting both to lower

Reply:
Thanks, was looking for something like this.